File: code_samples/lua/file_operations_reading.lua
-- File Operations: Reading
-- AI/ML Training Sample

Reading = {}
Reading.__ index = Reading

function Reading:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function Reading:process(input)
    self.data = input
end

function Reading:getData()
    return self.data
end

function Reading:validate()
    return #self.data > 0
end

-- Example usage
local instance = Reading:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/csharp/file_operations_reading.cs
using System;

/// <summary>
/// File Operations: Reading
/// AI/ML Training Sample
/// </summary>
public class Reading
{
    private string data;
    
    public Reading()
    {
        this.data = string.Empty;
    }
    
    public void Process(string input)
    {
        this.data = input;
    }
    
    public string GetData()
    {
        return this.data;
    }
    
    public bool Validate()
    {
        return !string.IsNullOrEmpty(this.data);
    }
    
    public static void Main(string[] args)
    {
        var instance = new Reading();
        instance.Process("example");
        Console.WriteLine($"Data: {instance.GetData()}");
        Console.WriteLine($"Valid: {instance.Validate()}");
    }
}
